Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp

```diff
@@ -382,6 +389,21 @@ static void webkit_website_data_manager_class_init(WebKitWebsiteDataManagerClass
             _("Whether the WebKitWebsiteDataManager is ephemeral"),
             FALSE,
             static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY)));
+
+    /**
+     * WebKitWebsiteDataManager:local-storage-quota:
+     *
+     * Quota for local storage (in bytes)
+     *
+     */
+    g_object_class_install_property(
+        gObjectClass,
+        PROP_LOCAL_STORAGE_QUOTA,
+        g_param_spec_uint("local-storage-quota",
+            _("Local storage quota"),
+            _("The maximum size of local storage in bytes"),
+            1, G_MAXUINT, 5 * 1024 * 1024,
+            static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY)));
```

Comment:
What's your idea about how this feature should work? Do you want to be able to change its value after the object gets created? Or should it be something that's only sent once when the object is created and it's never modified? For me it should be the latter: a quota is set for the WebKitWebsiteDataManager for the whole execution. Bear in mind that, as you're proposing the implementation, changes to the property are not propagated to existent NetworkProcess instances, just to those created af the value is set.
I would define the property as WEBKIT_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY, so it's defined only when the object is created and cannot be changed after that. And if this is done, the API methods webkit_website_data_manager_get_local_storage_quota and webkit_website_data_manager_set_local_storage_quota are not needed.

Reply:
Hi @magomez, yes the requirement is to set it once on initialization. I will remove the APIs and update the property flags.